packages/aws-cdk-lib/aws-s3/lib/notifications-resource/notifications-resource-handler.ts

```diff
@@ -63,14 +63,20 @@ export class NotificationsResourceHandler extends Construct {
       assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
     });
 
-    this.role.addManagedPolicy(
-      iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
-    );
     this.role.addToPrincipalPolicy(new iam.PolicyStatement({
       actions: ['s3:PutBucketNotification'],
       resources: ['*'],
     }));
 
+    if (props.role) {
+      cdk.Annotations.of(this).addWarningV2('@aws-cdk-lib/aws-s3:missingLambdaExecutionRole', 'CDK doesn\'t add service-roles if `notificationsHandlerRole` is provided, Make Sure Imported Role have AWSLambdaBasicExecutionRole added');
+    }
+    if (!props.role) {
```

Comment:
I'm not sure whether the conditions are correct. What happens when a user passed a Role construct? e.g. 
```ts
{
    role: new iam.Role(this, 'Role', {
      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
    })
}
```

This should just add a managed policy to the role.

Reply:
Thanks for the clarification. Do we really need two warning for this? I feel one warning is enough. Then we can keep the current code as-is for this file.